rust/src/elem_restriction.rs

```diff
@@ -298,12 +298,12 @@ impl ElemRestriction {
     /// let r = ceed.elem_restriction(nelem, 2, 1, compstride, nelem + 1, MemType::Host, &ind);
     ///
     /// let c = r.comp_stride();
-    /// assert_eq!(c, compstride as i32, "Incorrect component stride");
+    /// assert_eq!(c, compstride, "Incorrect component stride");
     /// ```
-    pub fn comp_stride(&self) -> i32 {
+    pub fn comp_stride(&self) -> usize {
         let mut compstride = 0;
         unsafe { bind_ceed::CeedElemRestrictionGetCompStride(self.ptr, &mut compstride) };
-        compstride
+        compstride as usize
```

Comment:
why don't you need to do `usize::try_from(compstride).unwrap()` here?

Reply:
Yeah, the compiler should have caught that. That's a typo on my part

Reply:
Why should the compiler have caught it?

Reply:
Because 1) compstride needs to be an i32 for the CeedElemRestrictionGetCompStride call to actually be vaild and 2) a return value cannot use `as xxx`

Reply:
I think the compiler sees 1) through the type inference, I wasn't aware of 2). Thank you.